src/controller/custom_controller.py

```
@@ -29,3 +29,40 @@ async def paste_from_clipboard(browser: BrowserContext):
+            if file_upload_el is None:
+                return ActionResult(error=f'No file upload element found at index {index}')
+
+            try:
+                await file_upload_el.set_input_files(path)
+                msg = f'Successfully uploaded file to index {index}'
+                return ActionResult(extracted_content=msg)
+            except Exception as e:
+                return ActionResult(error=f'Failed to upload file to index {index}')
+
+        @self.registry.action('Close file dialog', requires_browser=True)
+        async def close_file_dialog(browser: BrowserContext):
```

Ruff (0.8.2)

Missing return type annotation for private function `close_file_dialog`

Add return type annotation: `None`

(ANN202)